src/cleanlab_tlm/utils/rag.py

```diff
@@ -0,0 +1,532 @@
+            This indicates that the query should be considered when performing this evaluation.
+            The identifier should match what you use in your criteria text.
+            For example, using "User Question" as the identifier means your criteria should refer to the query as "User Question".
+            Set to None if this evaluation doesn't consider the query. Default is None.
+        context_identifier (str, optional): Identifier string for the context input in the evaluation criteria.
+            This indicates that the retrieved context should be considered when performing this evaluation.
+            The identifier should match what you use in your criteria text.
+            For example, using "Retrieved Documents" as the identifier means your criteria should refer to the context as "Retrieved Documents".
+            Set to None if this evaluation doesn't consider the context. Default is None.
+        response_identifier (str, optional): Identifier string for the response input in the evaluation criteria.
+            This indicates that the generated response should be considered when performing this evaluation.
+            The identifier should match what you use in your criteria text.
+            For example, using "AI Answer" as the identifier means your criteria should refer to the response as "AI Answer".
+            Set to None if this evaluation doesn't consider the response. Default is None.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        criteria: str,
+        query_identifier: Optional[str] = None,
+        context_identifier: Optional[str] = None,
+        response_identifier: Optional[str] = None,
+    ):
+        """
+        lazydocs: ignore
+        """
+        # Validate that at least one identifier is specified
+        if query_identifier is None and context_identifier is None and response_identifier is None:
+            raise ValueError(
+                "At least one of query_identifier, context_identifier, or response_identifier must be specified."
+            )
+
+        self.name = name
+        self.criteria = criteria
+        self.query_identifier = query_identifier
+        self.context_identifier = context_identifier
+        self.response_identifier = response_identifier
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the Eval object in dictionary format.
+
+        Returns:
+            str: A dictionary-like string representation of the Eval object.
+        """
+        return (
+            f"{{\n"
+            f"    'name': '{self.name}',\n"
+            f"    'criteria': '{self.criteria}',\n"
+            f"    'query_identifier': {self.query_identifier!r},\n"
+            f"    'context_identifier': {self.context_identifier!r},\n"
+            f"    'response_identifier': {self.response_identifier!r}\n"
+            f"}}"
+        )
+
+
+_DEFAULT_EVALS = [
```

Comment:
[nit] I would swap the order of these two prompts, I think more users will be interested in `context_sufficiency` so should display that first